### PR TITLE
Grant application for Boris Boarman

### PR DESCRIPTION
**Pull request by imboogieman**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (USDC & DOT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by imboogieman** [comment]
I have read and hereby sign the Contributor License Agreement.

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Hi @imboogieman thanks for the application. Could you please save the rename the file with a .md extension so it is easier to read? Thanks.

File: applications/Boris Boarman. Web3 Grant Coach.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+### Overview
+
+Web3 space is well-known for fostering innovation, and existing crypto grant programs and funding options have given birth to many promising startups and initiatives.
+
+However,
+
+- the process of searching for, finding, and applying for the right crypto funding option is cumbersome and time-consuming for both applicants and reviewers.
+- Applicants struggle with identifying suitable grants and preparing high-quality applications,
+- while reviewers face the challenge of managing numerous submissions.
+- To our research about 85% of proposals get rejections either because of irrelevance, poor quality, however reviewer's lack of domain competence and even nepotism among foundations members are among the most common reasons as well.
```
Comment (keeganquigley) [inline applications/Boris Boarman. Web3 Grant Coach.md:19]: Could you provide the source here? Is this from the State of Web3 Grants Report?

File: applications/Boris Boarman. Web3 Grant Coach.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+- An overview of the technology stack to be used:
+Frameworks:
+React.js: For building an interactive, dynamic UI.
+Next.js (optional): For server-side rendering and SEO optimization.
+Node.js with Express: For scalable, lightweight REST API development.
+Python with FastAPI: If Python is preferred for handling AI-based scoring.
+
+Wallet Integration:
+Web3Modal or RainbowKit: For integrating wallets like MetaMask, WalletConnect, etc.
```
Comment (keeganquigley) [inline applications/Boris Boarman. Web3 Grant Coach.md:46]: Do you plan for integration of Polkadot wallets here as well, such as Talisman, Nova Wallet, Subwallet, etc?

File: applications/Boris Boarman. Web3 Grant Coach.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+Libraries: ethers.js for blockchain interaction in the browser.
+
+NLP Models:
+OpenAI GPT Models: For proposal analysis and scoring.
+Hugging Face Models: For fine-tuned, on-premise scoring.
+InvestML: For impact analysis and projections. 
+
+Blockchain Interaction:
+ethers.js: For blockchain interaction.
+Alchemy, Infura, or a self-hosted Ethereum node: For connecting to blockchain networks.
```
Comment (keeganquigley) [inline applications/Boris Boarman. Web3 Grant Coach.md:56]: This seems to be left over from a previous application, as Polkadot doesn't use these RPCs...

File: applications/Boris Boarman. Web3 Grant Coach.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+- **Costs:** 34,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License	| Apache 2.0 / GPLv3 / MIT |
+| **0b.** | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can use Boris Boarman to validate and score proposals. |
+| **0c.** | Testing and Testing Guide |	Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests.|
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone.|
+| 0e. | Article | We will publish an article that explains what was done/achieved as part of the grant. |
+| 1. | Web App Graphic User Inerface | Main layout (chat, parameters sections, sessions), Problem Statement, Feasibility Plan, Innovation and Differentiation, Impact Potential, Team Profile, Milestones, Timelines, and Budget (Delivery Plan), Documentation download, User profile (registration, settings). |
```
Comment (keeganquigley) [inline applications/Boris Boarman. Web3 Grant Coach.md:183]: We only fund technical development, so we likely wouldn't fund some of these initial website setup deliverables. Additionally, we ask for some initial mockups/wireframes to be provided in the application for the front-end parts.

**Review by keeganquigley — CHANGES_REQUESTED** [review]

File: applications/Boris Boarman. Web3 Grant Coach.md (hunk 1 above)
Comment (imboogieman) [reply applications/Boris Boarman. Web3 Grant Coach.md:19]: @keeganquigley Certainly. 

Source 1. 
"Even as programs evolve, many still struggle with maintaining clear communication and
comprehensive documentation. It can be hard for grantees and community members to find
the information they need, which can lead to confusion and missed opportunities. " - source "State of Web3 Grants Report 2024" page 158.

Source 2.
"... a small group of established teams repeatedly winning funding, opportunities, and attention..." - source ["Are Web3 Grant Programs Truly Doing Their Job?"](https://hackernoon.com/are-web3-grant-programs-truly-doing-their-job) by Marianna Charalambous. 

Source 3.
"Grantees are overwhelmed by managing and promoting multiple grants across fragmented platforms, while funders struggle with an influx of proposals and high computational costs in evaluating them. " [source. ](https://gov.gitcoin.co/t/web3-funding-fatigue-a-growing-problem/19423)

Source 4.
"Web3 promised decentralization, but nepotism rules. I see it daily in grants and partnerships." - source [zktrivo's twitter](https://x.com/zktrivo/status/1841872024521588792), teal lead @ Mina Protocol.

Source 5. 
"Web3 grant programs are broken - we hate them." - source [burnt_xion's twitter](https://x.com/burnt_xion/status/1867577961555407263)


Source 6. 
"Nepotism is surely live and well in the web3 space" - source [RussLive215's twitter.](https://x.com/RussLive215/status/1834481166058971188)

Source 7.
"Nepotism or cronyism is a critical issue, as the expectation is that grants are given to the best researchers, and not to applicants that are socially, organizationally, or topic-wise nearby the decision-makers." - [source "Do interests affect grant application success? The role of organizational proximity"](https://www.researchgate.net/publication/344461606_Do_interests_affect_grant_application_success_The_role_of_organizational_proximity)

File: applications/Boris Boarman. Web3 Grant Coach.md (hunk 2 above)
Comment (imboogieman) [reply applications/Boris Boarman. Web3 Grant Coach.md:46]: @keeganquigley Trust Wallet supports Polkadot (DOT). You can securely store, manage, and stake your DOT tokens directly within the Trust Wallet app.

We have plans to support as much wallets as possible. For the first version we decided to select the most easiest to integrate to our minds as the integration of those wallets bears less technical risks as each new wallet integration requires technical review and a separately allocated resources.

File: applications/Boris Boarman. Web3 Grant Coach.md (hunk 3 above)
Comment (imboogieman) [reply applications/Boris Boarman. Web3 Grant Coach.md]: Updated in the file. Thank you.

File: applications/Boris Boarman. Web3 Grant Coach.md (hunk 4 above)
Comment (imboogieman) [reply applications/Boris Boarman. Web3 Grant Coach.md:183]: @keeganquigley The web application is the GUI for the solution we develop. The UI mockups can be seen here [Boris Boarman UI Mockups](https://drive.google.com/drive/folders/1QpLKJmFSAIPPsO6HUDtNGikW4ca2T08r?usp=sharing).

Shared the link in the original application, but it might have be missed. Update the file as well.

**Comment by keeganquigley** [comment]
Thanks for the changes @imboogieman I could see this possibly being useful as a widget for grant applications, but personally I think it is too expensive. I'd rather see a level 2 PoC first to determine how feasible the solution is. 

Additionally, I'm not sure how we would be able to integrate this into our W3F grants program. For one, we have private grants in addition to public ones, which we wouldn't be able to parse with this tool. But even for the public ones, I think the app would have to be in the form of a GitHub bot or something. Otherwise, the way your current front-end is set up, we would have to manually copy and paste applications into the app to be processed. This to me doesn't seem any different than using existing AI solutions. Therefore I won't approve it but I will mark it as ready for review to see if anyone else has other comments.

**Comment by imboogieman** [comment]
>I could see this possibly being useful as a widget for grant applications.

The first target audience is the applicants, so they can faster write and score their ideas, so the reviewers get less workload. The end goal is provider as a solution for DAO governance as well.

**Comment by imboogieman** [comment]
> but personally I think it is too expensive.

We estimated both the LLM testing and configuration activities for each of the 7 parameters. We put a detailed estimation. We calculated everything at minimum rates accounting risks and testing the model. 

Did you chance to check the spreadsheet with the estimation break-down?

**Comment by keeganquigley** [comment]
Thanks for your thoughts @imboogieman that's also part of my reason for rejecting it, because I believe the committee will want to fund a proposal generator for applicants. Not that AI can't be helpful, but in my opinion teams are overly reliant on it already, and I don't want to pay them to use it :)

But I will keep the proposal open for a few days for others to comment. Also I don't think the idea is a bad one, I just don't think it should be funded by W3F.

**Comment by imboogieman** [comment]
> I'd rather see a level 2 PoC first to determine how feasible the solution is. 

It is feasible as we have already developed the prototype with the 3 parameters and basic scoring model. https://boris-boarman-main-app.dep.zsoft.link/

And we are heading to have 7 parameters in total for the next version and more advanced scoring model with a range form 0 to 1000 points for better accuracy.

**Comment by keeganquigley** [comment]
> Did you chance to check the spreadsheet with the estimation break-down?

Thanks @imboogieman no I missed this so I will check it out.

**Comment by imboogieman** [comment]
> This to me doesn't seem any different than using existing AI solutions.

Is it possible to share what solution you meant? As OpenAI and other providers' default LLMs are not proving accurate results when it comes to scoring thus we are using open model just for reasoning. And for numeric scoring we have our own scoring rubric and built-in  module. 

And even now we are having from 5 to 10 points range variance that is why testing and fine-tuning is a big part of the development to reach more or less accurate results. For the next version we will be using Invest ML, a model specifically trained for investment decision, for now it will over complicate the development and increase the cost significantly.

**Comment by imboogieman** [comment]
> > Did you chance to check the spreadsheet with the estimation break-down?
> 
> 
> 
> Thanks @imboogieman no I missed this so I will check it out.

The tab "Cost break-down" and "Estimation" 

https://docs.google.com/file/d/1n4CuKghjtqOstTCBLnVJDUUf1dkN8EcZ/edit

I checked the estimation for a few times to make it as accurate as possible, though I still think we will go over as there is a lot of unknown. However, this approach, I consider much more transparent and accurate. Also, we want to implement milestone estimation as well, so the development is priced evenly. That is another important feature of the product and AI in this case as estimation takes time and a lot of developers are not able to prologues accurate estimates.

**Comment by imboogieman** [comment]
> Also because it's not super specific to Polkadot.

As we discovered Polkadot is currently running numerous grant programs for the ecosystem products 

https://polkadot.com/blog/the-ultimate-2024-polkadot-grants-and-funding-guide

Our solution also is able to select the most appropriate grant program for an idea or initiative. I'm personally busy now with this recommendation functionality at the moment.

**Comment by keeganquigley** [comment]
ok thanks for your feedback @imboogieman I'll bring it up for internal discussion in our sync tomorrow morning and ask the rest of the committee to take a look.

**Comment by imboogieman** [comment]
> ok thanks for your feedback @imboogieman I'll bring it up for internal discussion in our sync tomorrow morning and ask the rest of the committee to take a look.

Hey @keeganquigley ! How is it going? Did you chance to bring other reviewers and get their opinions?

Just to support even more our application, here I want to share some quotes.

_"AI can solve this by voting in place of humans. Instead of a human making a voting decision and signing a transaction with their wallet, they can delegate their tokens to an AI agent to vote on behalf of them."_ - source [Aragon blog](https://www.aragon.org/how-to/the-future-of-daos-is-powered-by-ai).

_"Once the proposal is created, a DAO can start the voting process once the proposal is submitted. While the voting commences, AI agents can act on behalf of humans."_ - source ["Artificial Intelligence and Decentralized Autonomous Organizations: Where two worlds meet", a blog post of Wipro Tech, Oct 24, 2024](https://wiprotechblogs.medium.com/artificial-intelligence-and-decentralized-autonomous-organizations-where-two-worlds-meet-2173312ae764)

As the next steps we will provide APIs to DAOs so they can integrate this solution into their decision-making process. These two milestones are the first steps to test all the hypothesis and find and fine-tune the best models. 

Imagine now it is something small that can grow into a real big and essential thing for the whole web3 community.

**Comment by imboogieman** [comment]
> Thanks again for the application, @imboogieman, b after careful consideration, the committee decided not to support it. There was mainly concern about the usefulness of an LLM with regard to grant applications and the price tag. 
> 
> 
> 
> However, we hope this won't deter you and you'll find a way to make your project a reality. Please check our page on [alternative funding opportunities](https://grants.web3.foundation/docs/funding). Best of luck!

Hi, Samuel! 

Is it possible you can provide a more detailed feedback? As I don't understand what concerns you have as our research shows that even other foundations are repeating our idea https://www.aipgf.com/. 

As well as we provided several links to the trusted sources highlighting the problem. As well quotes from trusted founders that such a solution can help DAOs a lot. It is just embarrassing. 

As to the price tag? Will you be likely to approve if we shrink the scope to fit the cost under Option 1 for example?

**Comment by imboogieman** [comment]
> Thanks again for the application, @imboogieman, b after careful consideration, the committee decided not to support it. There was mainly concern about the usefulness of an LLM with regard to grant applications and the price tag.
> 
> However, we hope this won't deter you and you'll find a way to make your project a reality. Please check our page on [alternative funding opportunities](https://grants.web3.foundation/docs/funding). Best of luck!


https://x.com/web3dealdesk

It is easy to steel than to support. This is it all about. Now I understood why it took so long and why you reject the proposal without any clear and logic feedback. So silly. I'm taking this case to the media, so let the people know who is behind the web3 foundation claiming to build honest and transparent internet.  Just another prove that the idea is right to the problem.

**Comment by semuelle** [comment]
> Is it possible you can provide a more detailed feedback?

Yes.

- We have looked into using LLMs for our internal purposes a while ago and found that they were too inaccurate. But even if they weren't, there is so much contextual, ever-changing information beyond the application doc necessary that it would be infeasible to build and maintain a sufficiently accurate model for our purposes at this point. 
- We are not keen on receiving even more LLM-written grant applications, as the parts that LLMs are good in are the least interesting bits of an application and at worst, mask incompetence of the applicant.
- We prefer to talk to potential applicants directly about funding mechanisms, as it will help us get a better picture of what they need and where they are in their journey. 
- Funding opportunities come and go, so what may have been right six weeks ago might not be right, or even exist, today. Your training data will be outdated before you collected enough. 
- You did not explain what data and functionality the smart contracts would be responsible for, where you would deploy them, or address the fact that most parachains do not support Solidity smart contracts, excluding them from the concept. 
- None of your deliverables even mentions Polkadot or anything related to it. Perhaps this is due to the fact that there is indeed little connection to _any_ blockchain, as evidenced by your earlier applications in other ecosystems.

To conclude: we might have looked at different data, or looked at them through different lenses, just like the organisations that do support similar projects did. I'd be happy to be proven wrong. However, I do not see sufficient "research" in your document that would warrant our conclusion "embarrassing". That being said, you linked to the `State of Web3 Grants` report three times and called it "provid[ing] several links to trusted sources", so I'm skeptical our ideas of what constitutes research overlap.
